server/command_handler.go

```diff
@@ -44,13 +50,122 @@ func subscribeHandler(server *Server, client *Client, frm *frame.Frame) {
 		sendError(client, nil, "destination field is required in the subscribe frame")
 		return
 	}
-	if _, ok := server.subscribers[destination]; !ok {
+	if _, ok := server.destinations[destination]; !ok {
 		sendError(client, nil, destination+"is not supported in the server")
 		return
 	}
-	server.subscribersLock.Lock()
-	subList := server.subscribers[destination]
-	newSubList := append(subList, client)
-	server.subscribers[destination] = newSubList
-	server.subscribersLock.Unlock()
+	newSubscriber := &subscriber{
+		client:       client,
+		isSubscribed: true,
+	}
+	server.destinationsLock.Lock()
+	subList := server.destinations[destination]
+	newSubList := append(subList, newSubscriber)
+	server.destinations[destination] = newSubList
+	server.destinationsLock.Unlock()
```

Comment:
NIT;
better practice to use unlock as _defered call_ after you acquire the lock. with this you are certain that the lock will be freed even if there is a panic or an error.